components/status_dropdown/status_dropdown.jsx

```diff
@@ -37,35 +41,101 @@ export default class StatusDropdown extends React.PureComponent {
         status: UserStatuses.OFFLINE,
     }
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            openUp: false,
+            width: 0,
+        };
+    }
+
+    getCurrentDateTime = (tz, enable) => {
+        let currentDate;
+        if (enable) {
+            if (tz.useAutomaticTimezone) {
+                currentDate = getCurrentDateTimeForTimezone(tz.automaticTimezone);
+            } else {
+                currentDate = getCurrentDateTimeForTimezone(tz.manualTimezone);
+            }
+        }
+        return currentDate;
+    }
+
+    dndTimes = ['30 mins', '1 hour', '2 hours', 'Today', 'Tomorrow', 'Custom']
```

Comment:
We could probably move this to a constant file. I am 1/5 this would be a good addition in `UserStatuses`